src/System.Configuration.ConfigurationManager/src/System/Configuration/ImplicitMachineConfigHost.cs

```diff
@@ -23,14 +23,14 @@ internal ImplicitMachineConfigHost(IInternalConfigHost host)
             out string locationConfigPath, IInternalConfigRoot configRoot, params object[] hostInitConfigurationParams)
         {
             // Stash the filemap so we can see if the machine config was explicitly specified
-            _fileMap = (ConfigurationFileMap)hostInitConfigurationParams[0];
+            _fileMap = hostInitConfigurationParams[0] as ConfigurationFileMap;
```

Comment:
Please upstream the change if it really works (the value can be null)

Reply:
OK - the interesting part here is that System.Web has a different order to hostInitConfigurationParams.  Hence the need to use as.